src/Doctrine/AbstractElasticaToModelTransformer.php

```diff
@@ -95,12 +95,16 @@ public function transform(array $elasticaObjects)
         $objects = $this->findByIdentifiers($ids, $this->options['hydrate']);
         $objectsCnt = count($objects);
         $elasticaObjectsCnt = count($elasticaObjects);
+        $propertyAccessor = $this->propertyAccessor;
+        $identifier = $this->options['identifier'];
         if (!$this->options['ignore_missing'] && $objectsCnt < $elasticaObjectsCnt) {
-            throw new \RuntimeException(sprintf('Cannot find corresponding Doctrine objects (%d) for all Elastica results (%d). IDs: %s', $objectsCnt, $elasticaObjectsCnt, implode(', ', $ids)));
+            $missingIds = array_diff($ids, array_map(function ($object) use ($propertyAccessor, $identifier) {
+                return $propertyAccessor->getValue($object, $identifier);
+            }, $objects));
+
+            throw new \RuntimeException(sprintf('Cannot find corresponding Doctrine objects (%d) for all Elastica results (%d). Missing IDs: %s. IDs: %s', $objectsCnt, $elasticaObjectsCnt, implode(', ', $missingIds), implode(', ', $ids)));
```

Comment:
Maybe we should find a better wording for the exception message, but I'm not the best for this king of things.